File: 0872_Leaf-Similar Trees/Leaf-SimilarTrees.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool LeafSimilar(TreeNode root1, TreeNode root2) {
        var list1 = new List<int>();
        var list2 = new List<int>();
        
        collectLeafs(root1, list1);
        collectLeafs(root2, list2);
        
        if(list1.Count != list2.Count) return false;
        for(int i=0;i<list1.Count;i++){
            if(list1[i] != list2[i]) return false;
        }
        
        return true;
    }
    
    private void collectLeafs(TreeNode node, List<int> list){
        if(node == null) return;
        if(node.left == null && node.right == null) list.Add(node.val);
        collectLeafs(node.left, list);
        collectLeafs(node.right, list);
    }
}

File: 0872_Leaf-Similar Trees/Leaf-SimilarTrees.java
class Solution {
    public boolean leafSimilar(TreeNode root1, TreeNode root2) {
        List<Integer> list1 = new ArrayList<Integer>();
        List<Integer> list2 = new ArrayList<Integer>();
        traverse(root1, list1);
        traverse(root2, list2);
        
        if(list1.size() != list2.size()) return false;
        for(int i=0;i<list1.size();i++)
            if(list1.get(i) != list2.get(i)) return false;
        
        return true;
    }
    
    private void traverse(TreeNode node, List<Integer> list) {
        if(node == null) return;
        if(node.left == null && node.right == null) list.add(node.val);
        traverse(node.left, list);
        traverse(node.right, list);
    }
}